### DISCUSSION TITLE
Error with reference and displaced meshes

### DISCUSSION THREAD

**Discussion by rtaylo45** [idaholab/moose#18736] [Q&A General] [answered]
Hello,

I was wondering if someone could give me some insight into the following error im getting 
```
*** ERROR ***
Our reference and displaced meshes are not in sync with respect to whether we should delete remote elements.
```

Ive built a moose app that is coupled to another moose app which uses a displaced mesh. Not sure if it has something to do with that. This error only seems to be affecting some of my unit test and their doesn't appear to be a specific reason why. 

Thanks
Zack

**Comment by GiudGiud** [comment, marked as answer]
Hello

We're going to need more context on this. Are you deleting elements? Through what mechanism? (distributed mesh?)
The code triggering this
```
  else if (_current_task == "delete_remote_elements_after_late_geometric_ghosting")
  {
    TIME_SECTION("deleteRemoteElems", 2, "Deleting Remote Elements");

    if (_displaced_mesh &&
        (_mesh->needsRemoteElemDeletion() != _displaced_mesh->needsRemoteElemDeletion()))
      mooseError("Our reference and displaced meshes are not in sync with respect to whether we "
                 "should delete remote elements.");
```
seems to show that one of the subapp or main app is wanting to delete some elements and the other not

Guillaume

**Reply by rtaylo45** [reply]
Its being triggered by trying to add periodic boundary conditions using this block
```
[BCs]
  [Periodic]
    [all]
      variable = 'c_i'
      auto_direction = 'x y'
    []
  []
[]
```

**Reply by GiudGiud** [reply]
Oh sorry I missed this. 
@fdkong any idea? Who s our expert on periodic BCs?

**Reply by fdkong** [reply]
@rtaylo45 Are you able to provide a minimal example on this? 

It simply stated that two meshes are out of sync. Did you change any moose code?

**Reply by rtaylo45** [reply]
This error came up from coupling two codes together. It turns out that we don't need to use periodic BCs when the codes run in this way and this error came up from a unit test in one of the codes. To work around this the unit test were placed in groups and for the special case of the codes being compiled together this group of test is not needed and so they are not run. Thanks for the follow up though.
